src/shared/scrapers/USA/WV/index.js

```diff
@@ -77,43 +77,85 @@ const scraper = {
     'Wyoming County'
   ],
 
-  async scraper() {
-    const $ = await fetch.page(this.url);
+  scraper: {
+    '0': async function() {
+      const $ = await fetch.page(this.url);
 
-    const $p = $('p:contains("Counties with positive cases")');
+      const $p = $('p:contains("Counties with positive cases")');
 
-    const list = $p
-      .text()
-      .split(':')[1]
-      .split(',');
+      const list = $p
+        .text()
+        .split(':')[1]
+        .split(',');
 
-    let counties = [];
+      let counties = [];
 
-    for (const item of list) {
-      const items = item.split('(');
-      if (items.length !== 2) {
-        continue;
-      }
+      for (const item of list) {
+        const items = item.split('(');
+        if (items.length !== 2) {
+          continue;
+        }
+
+        const county = geography.addCounty(parse.string(items[0]));
+        let cases = items[1];
 
-      const county = geography.addCounty(parse.string(items[0]));
-      let cases = items[1];
+        if (cases.slice(-1) !== ')') {
+          continue;
+        }
 
-      if (cases.slice(-1) !== ')') {
-        continue;
+        cases = parse.number(cases.slice(0, -1));
+
+        counties.push({
+          county,
+          cases
+        });
       }
 
-      cases = parse.number(cases.slice(0, -1));
+      counties = geography.addEmptyRegions(counties, this._counties, 'county');
+      counties.push(transform.sumData(counties));
 
-      counties.push({
-        county,
-        cases
-      });
-    }
+      return counties;
+    },
+    '2020-04-03': async function() {
+      this.url =
+        'https://wabi-us-gov-virginia-api.analysis.usgovcloudapi.net/public/reports/187b4de8-78ef-40be-9510-7fa9a1ef89f2/modelsAndExploration';
+      const options = {
+        headers: { 'X-PowerBI-ResourceKey': '187b4de8-78ef-40be-9510-7fa9a1ef89f2' }
+      };
+      const data = await fetch.json(this.url, undefined, options);
+      const { sections } = data.exploration;
 
-    counties = geography.addEmptyRegions(counties, this._counties, 'county');
-    counties.push(transform.sumData(counties));
+      let counties = [];
 
-    return counties;
+      // This is all insanely horrible
+      const heading = 'CONFIRMED CASES PER COUNTY:';
+      for (const sec of sections) {
+        for (const vc of sec.visualContainers) {
+          if (vc.config.indexOf(heading) !== -1) {
+            const config = JSON.parse(vc.config);
+            let { textRuns } = config.singleVisual.objects.general[0].properties.paragraphs[0];
+            textRuns = textRuns.map(tr => parse.string(tr.value));
+            if (textRuns[0] !== heading) {
+              throw new Error('Unknown json layout');
+            }
+            const pieces = textRuns[1].split(',');
+            for (const row of pieces) {
```

Comment:
Do you think it's useful to regex the content of pieces, or of row, for sanity?

Reply:
Using a regex on the row might be more pleasant yes. But it'll be putting lipstick on a pig.

I don't know js regex syntax, so I used the hammer I knew.